### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by KittNeKit**

### CODE REVIEW

File: services/movie.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from db.models import Movie
+
+
+def get_movies(
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None) -> Movie:
```
Comment (Bohdan-Salamakha) [inline services/movie.py:6]: invalid return typehint
+
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contains string elements and can be None has typehint Optional[list[str]]

File: services/movie.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
+    return Movie.objects.get(id=movie_id)
+
+
+def create_movie(movie_title: str,
+                 movie_description: str,
+                 genres_ids: list[int] = None,
+                 actors_ids: list[int] = None) -> Movie:
```
Comment (Bohdan-Salamakha) [inline services/movie.py:24]: invalid return typehint
+
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contains string elements and can be None has typehint Optional[list[str]]

File: services/movie_session.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+from typing import Optional
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int) -> MovieSession:
```
Comment (Bohdan-Salamakha) [inline services/movie_session.py:10]:
```suggestion
def create_movie_session(
        movie_show_time: datetime,
        movie_id: int,
        cinema_hall_id: int
        ) -> MovieSession:
```

File: services/movie_session.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+
+def get_movies_sessions(session_date: datetime = None) -> MovieSession:
```
Comment (Bohdan-Salamakha) [inline services/movie_session.py:18]: invalid return typehint
+
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contains string elements and can be None has typehint Optional[list[str]]

File: services/movie_session.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int,
+                         show_time: Optional[str] = None,
+                         movie_id: Optional[int] = None,
+                         cinema_hall_id: Optional[int] = None) -> MovieSession:
```
Comment (Bohdan-Salamakha) [inline services/movie_session.py:32]: invalid return typehint

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

File: services/movie.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+from typing import Optional
+
+from db.models import Movie
+
+
+def get_movies(
+        genres_ids: Optional[list[int]] = None,
+        actors_ids: Optional[list[int]] = None) -> Movie:
```
Comment (Bohdan-Salamakha) [inline services/movie.py:8]:
```suggestion
def get_movies(
        genres_ids: Optional[list[int]] = None,
        actors_ids: Optional[list[int]] = None
        ) -> Movie:
```

Comment (Bohdan-Salamakha) [inline services/movie.py:8]: invalid return typehint

File: services/movie.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
+    return Movie.objects.get(id=movie_id)
+
+
+def create_movie(movie_title: str,
+                 movie_description: str,
+                 genres_ids: Optional[list[int]] = None,
+                 actors_ids: Optional[list[int]] = None) -> Movie:
```
Comment (Bohdan-Salamakha) [inline services/movie.py:26]: invalid return typehint

File: services/movie_session.py (hunk 4 above)
Comment (Bohdan-Salamakha) [reply services/movie_session.py:21]: not fixed

File: services/movie_session.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+
+def get_movies_sessions(session_date: datetime = None) -> MovieSession:
+    queryset = MovieSession.objects.all()
+    if session_date is not None:
+        queryset = MovieSession.objects.filter(show_time__date=session_date)
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: Optional[int]) -> QuerySet:
```
Comment (Bohdan-Salamakha) [inline services/movie_session.py:28]: invalid return typehint

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

File: services/movie.py (hunk 6 above)
Comment (YukihiroSM) [reply services/movie.py:8]: Not fixed

Comment (YukihiroSM) [reply services/movie.py:8]: Not fixed

File: services/movie.py (hunk 7 above)
Comment (YukihiroSM) [reply services/movie.py:26]: not fixed. The function doesn't return Movie.

File: services/movie_session.py (hunk 4 above)
Comment (YukihiroSM) [reply services/movie_session.py:19]: still not fixed :) MoviesSession is not being returned by this function

**Review by YukihiroSM — CHANGES_REQUESTED** [review]

File: services/movie.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
+    return Movie.objects.get(id=movie_id)
+
+
+def create_movie(movie_title: str,
+                 movie_description: str,
+                 genres_ids: Optional[list[int]] = None,
+                 actors_ids: Optional[list[int]] = None) -> QuerySet:
```
Comment (olena-hul) [inline services/movie.py:28]: Return annotation is wrong, it returns None

File: services/movie_session.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    if show_time is not None:
+        MovieSession.objects.filter(id=session_id).update(show_time=show_time)
+    if movie_id is not None:
+        MovieSession.objects.filter(id=session_id).update(movie_id=movie_id)
+    if cinema_hall_id is not None:
+        MovieSession.objects.filter(id=session_id).update(
+            cinema_hall_id=cinema_hall_id)
+
+
+def delete_movie_session_by_id(session_id: int) -> MovieSession:
```
Comment (olena-hul) [inline services/movie_session.py:46]: It does not return MovieSession. `.delete` returns the number of objects deleted

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: services/movie.py (hunk 9 above)
Comment (olena-hul) [reply services/movie.py:28]: Comments are not fixed

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: services/movie_session.py (hunk 10 above)
Comment (olena-hul) [reply services/movie_session.py:46]: Not fixed

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by olena-hul — APPROVED** [review]

**Pull request closed without merging**